Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -1085,4 +1090,27 @@ HackStudioWidget::~HackStudioWidget()
     }
 }
 
+bool HackStudioWidget::request_close()
+{
+    if (!any_document_is_dirty())
+        return true;
+
+    auto result = GUI::MessageBox::show(window(), "There are unsaved changes, do you really want to exit?", "Unsaved changes", GUI::MessageBox::Type::Warning, GUI::MessageBox::InputType::YesNoCancel);
+
+    if (result == GUI::MessageBox::ExecYes)
+        return true;
+
+    return false;
```

Comment:
If "Yes" or "Not Yes" are the choices that matter, the message box should be a Yes/No box, instead of a Yes/No/Cancel box.

Reply:
"There are unsaved changes, do you want to save before exiting?" with a "Yes" (save and exit), "No" (dont save and exit) and "Cancel" (dont save and dont exit) is how most programs i've used in the past have done it, so replacing this with that might be nice :)

Reply:
Yes, I totally agree @IdanHo. Please see my changes.